src/SAWScript/JavaPretty.hs

```diff
@@ -6,69 +6,75 @@ Maintainer  : atomb
 Stability   : provisional
 -}
 
+{-# LANGUAGE OverloadedStrings #-}
+
 module SAWScript.JavaPretty where
 
 import Data.Maybe (fromMaybe)
 
-import Text.PrettyPrint.ANSI.Leijen hiding ((<$>))
+import Prettyprinter
 
 import Language.JVM.Common
 
 import Verifier.Java.Codebase
 
-prettyClass :: Class -> Doc
+prettyClass :: Class -> Doc ann
 prettyClass cls = vcat $
-  [ empty
-  , text "Class name:" <+> text (unClassName (className cls)) <+>
+  [ emptyDoc
+  , "Class name:" <+> prettyClassName (className cls) <+>
     parens (commas attrs)
-  , text "Superclass:" <+> text (fromMaybe "none" (fmap unClassName (superClass cls)))
-  , empty
+  , "Superclass:" <+> maybe "none" prettyClassName (superClass cls)
+  , emptyDoc
   ] ++
   (if null (classInterfaces cls)
       then []
-      else [ text "Interfaces:"
-           , indent 2 (vcat (map (text . unClassName) (classInterfaces cls)))
-           , empty
+      else [ "Interfaces:"
+           , indent 2 (vcat (map prettyClassName (classInterfaces cls)))
+           , emptyDoc
            ]) ++
-  [ text "Fields:"
+  [ "Fields:"
   , indent 2 (vcat (map prettyField (classFields cls)))
-  , empty
-  , text "Methods:"
+  , emptyDoc
+  , "Methods:"
   , indent 2 (vcat (map prettyMethod (classMethods cls)))
-  , empty
+  , emptyDoc
   ]
   where attrs = concat
-          [ if classIsPublic cls then [text "public"] else []
-          , if classIsFinal cls then [text "final"] else []
-          , if classHasSuperAttribute cls then [text "super"] else []
-          , if classIsInterface cls then [text "interface"] else []
-          , if classIsAbstract cls then [text "abstract"] else []
+          [ if classIsPublic cls then ["public"] else []
+          , if classIsFinal cls then ["final"] else []
+          , if classHasSuperAttribute cls then ["super"] else []
+          , if classIsInterface cls then ["interface"] else []
+          , if classIsAbstract cls then ["abstract"] else []
           ]
 
-prettyField :: Field -> Doc
+-- FIXME: avoid unpacking via String
+prettyClassName :: ClassName -> Doc ann
+prettyClassName cname = pretty (unClassName cname)
+
+prettyField :: Field -> Doc ann
 prettyField f = hsep $
-  [ text (show (fieldVisibility f)) ] ++
+  [ viaShow (fieldVisibility f) ] ++
   attrs ++
-  [ text (show (ppType (fieldType f))) -- TODO: Ick. Different PPs.
-  , text (fieldName f)
+  [ viaShow (ppType (fieldType f)) -- TODO: Ick. Different PPs.
```

Comment:
In case this was (mostly) an automatic change: does this comment still apply?  If not is seems like the `viaShow` isn't needed. (One more below, line 77).

I the comment does still apply, should we have an issue open somewhere and reference it from this comment?

Reply:
Yes, that comment still applies. Function `ppType` comes from the `jvm-parser` package, which is still using the HughesPJ pretty printing library (package `pretty`), which has its own separate `Doc` type. The "Ick" used to be for converting from HughesPJ `Doc` to `ansi-wl-pprint` `Doc`, and now it's for going from HughesPJ to `prettyprinter` Doc. I'll open a ticket and mention this file.

Reply:
See #951.